test/extensions/filters/network/mysql_proxy/mysql_auth_plugin_test.cc

```diff
@@ -0,0 +1,170 @@
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "mysql_test_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MySQLProxy {
+
+/* the api of mysql old password function is */
+TEST(AuthHelperTest, OldPassWordHash) {
+  /*
+   * OldPasswordCases generated from MySQL Server by "set old_passwords=1; SELECT PASSWORD(text);"
+   */
+  std::map<std::string, std::string> oldPassWordCases = {{
+      {"lspbmbnl", "04634f346a43ff51"}, {"sxvpxvng", "0edb7f6258b0580a"},
+      {"npzsltxt", "4dd0c97d07484d41"}, {"izosmxhr", "33fe1bd20270bc72"},
+      {"mtemsuyl", "17614bd602386385"}, {"kcjypgul", "57257c4b0e224152"},
+      {"herxsyge", "201bb5bd30541f27"}, {"ftlpfzvm", "3a4f942d449b9061"},
+      {"jywqqfrh", "5960d6ce7f814ccc"}, {"omosqeyy", "4f5a09846cdd86ef"},
+      {"ncjjzjhh", "379f21fb7511644f"}, {"zzsmxelk", "7cd63d5e534ca948"},
+      {"lwqwfpve", "6e083fea5fe9717f"}, {"jdubrwik", "0fe9dc444a4259f3"},
+      {"cogoctit", "1974fdc23962d219"}, {"namdszhy", "76e89bdc5148efe0"},
+      {"zlkxtjuh", "2bb60d22103409ad"}, {"flhmfvwt", "7f5cd79c0a58d21f"},
+      {"ogwcccqb", "45890ffb79c2f03c"}, {"ynrwbuqm", "4b4e61473903c4d2"},
+      {"ipqwmcgj", "3698a07c25dc1a97"}, {"nljxsrqy", "3921fd114429a6e6"},
+      {"vzzlcyvm", "386bc3b7782c0379"}, {"vcjpgsfn", "19d31ddb4aa048be"},
+      {"ddvtbfhi", "0a426efb0ae239a3"}, {"ehdosgsu", "0512db8036b9d3de"},
+      {"exekfjbz", "41e0df8705cab80e"}, {"xialkbfn", "7b5f19d161bbbaef"},
+      {"jcpghhfo", "5aca66af047a78b5"}, {"qixefqlq", "574ec4711241808b"},
+      {"unlyxufj", "60f9761b33fb784f"}, {"difewfvw", "0500eb5443cdf0b5"},
+      {"viemvtrd", "39a422f716fa7d76"}, {"spgulhxe", "4f1bfbee7c7c43fd"},
+      {"pkealafr", "410bb4906c6ee405"}, {"uhwesfin", "4b211eeb0102007a"},
+      {"fdzgongj", "118415ab577e08e7"}, {"shjgdjul", "6a7ff4c95e7aeba4"},
+      {"spisholn", "00c6d81259d142d5"}, {"vojzqdew", "061f0c1c11d00b82"},
+      {"jpwivnyw", "114cea307c297a4d"}, {"kzdntqjb", "07ee40ca5724732e"},
+      {"yznplsgh", "153e4895794f87f0"}, {"wvzbfusy", "5b9b138e364a3cf3"},
+      {"cgaxolxb", "6e2d938a0806323e"}, {"tlpsvudx", "2b10508001af7188"},
+      {"ptmscvsq", "35c03bcb493e24ef"}, {"imyjbide", "7f5335df7d9116a2"},
+      {"qmwrjnpk", "3dda289c041bc6d7"}, {"lmbnatik", "4b4eec946b3e3050"},
+  }};
+
+  for (const auto& old_password_case : oldPassWordCases) {
+    auto old_hash = OldPassword::hash(old_password_case.first);
+    EXPECT_EQ(old_password_case.second,
+              MySQLTestUtils::encodeUint32Hex(old_hash.data(), old_hash.size()));
+  }
+  EXPECT_EQ(OldPassword::hash("lmbnatik"), OldPassword::hash("lm bna  tik"));
+}
+
+TEST(AuthHelperTest, NativePassWordHash) {
+  /*
+   * NativePasswordCases generated from MySQL Server by "set old_passwords=0; SELECT
+   * PASSWORD(text);" "PASSWORD(text) = SHA1(SHA1(text))"
+   */
+  std::map<std::string, std::string> nativePassWordCases = {{
+      {"lspbmbnl", "448ecbe03d36ae7ae9fa0168e6a01a92abd8f7d6"},
```

Comment:
I wonder those sets are randomly generated or cover different cases?

Reply:
Yes, it's generated randomly. But it seems that one test case can represent the correctness. I will remove excess test cases.